packages/fuel-indexer/src/executor.rs

```diff
@@ -74,18 +75,16 @@ impl ExecutorSource {
 pub fn run_executor<T: 'static + Executor + Send + Sync>(
     fuel_node_addr: &str,
     mut executor: T,
-    start_block: Option<u64>,
+    start_block: &u64,
     kill_switch: Arc<AtomicBool>,
     stop_idle_indexers: bool,
 ) -> impl Future<Output = ()> {
-    let start_block_value = start_block.unwrap_or(1);
-    let mut next_cursor = if start_block_value > 1 {
-        let decremented = start_block_value - 1;
+    let mut next_cursor = if *start_block > 1 {
```

Comment:
Hm not too sure what this logic is doing, but I think `get_start_block` should return a value that doesn't have to be changed/altered by `run_executor` anymore 🤔